pkg/assembler/backends/testing/hasSBOM.go

```diff
@@ -110,21 +123,25 @@ func (c *demoClient) HasSBOMs(ctx context.Context, hasSBOMSpec *model.HasSBOMSpe
 		if uriMatchOrSkip && collectorMatchOrSkip && originMatchOrSkip {
 			if hasSBOMSpec.Package == nil && hasSBOMSpec.Source == nil {
 				collectedHasSBOM = append(collectedHasSBOM, h)
-			} else if hasSBOMSpec.Package != nil && h.Package != nil {
-				if hasSBOMSpec.Package.Type == nil || h.Package.Type == *hasSBOMSpec.Package.Type {
-					newPkg := filterPackageNamespace(h.Package, hasSBOMSpec.Package)
-					if newPkg != nil {
-						collectedHasSBOM = append(collectedHasSBOM, h)
+			} else if hasSBOMSpec.Package != nil {
```

Comment:
Can you do something similar like you did with uriMatchOrSkip, etc.? just all the nested ifs make it hard to grok exactly what's going on. Maybe even turn this set of things into a helper.

Reply:
yeah I can break them out more.